crates/node/src/service.rs

```diff
@@ -423,41 +423,26 @@ pub fn new_full(
     let (commitment_state_diff_tx, commitment_state_diff_rx) = mpsc::channel(5);
 
     // initialize data availability worker
-    if let Some((da_layer, da_path)) = da_layer {
+    if let Some((_da_layer, _da_path)) = da_layer {
```

Comment:
The fact they are not used it the signal that they should not be passed as arguments to the function.

I think you can remove this arg: `da_layer: Option<(DaLayer, PathBuf)>`
And just check `if let Some(da_client) = da_client` in order to know if we should run a da client

Reply:
if i remove da_layer from the function i guess i'll have to do in run.rs this part :
```
task_manager.spawn_essential_handle().spawn(
            "commitment-state-diff",
            Some("madara"),
            CommitmentStateDiffWorker::<_, _, StarknetHasher>::new(client.clone(), commitment_state_diff_tx)
                .for_each(|()| future::ready(())),
        );
```
        
but i won't have access to all the context like the taskManager, commitment_state_diff etc..., what do you advice ?

Reply:
Or maybe do you want me to run `service::new_full` only if there is a da_layer ?